src/transformState.js

```diff
@@ -4,8 +4,28 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
-}
+  for (const { type, extraData, keysToRemove } of actions) {
+    switch (type) {
+      case 'addProperties':
+        Object.assign(state, extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete state[key];
+        }
+        break;
 
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default: break;
```

Comment:
also it is good practice to make user know about error. log the message in the console, or throw the error